test/beacon/Voucher.test.ts

```diff
@@ -405,5 +408,155 @@ describe('Voucher', function () {
+                    .to.be.equal(voucherInitialCreditBalance - dealPrice)
+                    .to.be.equal(await getVoucherBalanceOnIexecPoco())
+                    .to.be.equal(voucherInitialSrlcBalance - dealPrice);
+                expect(await getRequesterBalanceOnIexecPoco()).to.be.equal(
+                    requesterInitialSrlcBalanceBefore,
+                );
+                expect(await voucher.getSponsoredAmount(dealId)).to.equal(dealPrice);
+            });
+
+            it('Should match orders boost with full non-sponsored amount', async () => {
+                const voucherInitialCreditBalance = await voucher.getBalance();
+                const voucherInitialSrlcBalance = await getVoucherBalanceOnIexecPoco();
+                expect(dealPrice).to.be.greaterThan(0); // just make sure the deal will not be free
+                // Deposit in iExec account of requester
+                await iexecPocoInstance.transfer(requester, dealPrice).then((tx) => tx.wait());
+                // Allow voucher to spend non-sponsored amount
+                await iexecPocoInstance
+                    .connect(requester)
+                    .approve(await voucher.getAddress(), dealPrice)
+                    .then((tx) => tx.wait());
+                const requesterInitialSrlcBalance = await getRequesterBalanceOnIexecPoco();
+
+                await expect(
+                    voucherWithOwnerSigner.matchOrdersBoost(
+                        appOrder,
+                        datasetOrder,
+                        workerpoolOrder,
+                        requestOrder,
+                    ),
+                )
+                    .to.emit(iexecPocoInstance, 'Transfer')
+                    .withArgs(requester.address, await voucher.getAddress(), dealPrice)
+                    .to.emit(voucher, 'OrdersBoostMatchedWithVoucher');
+                expect(await voucher.getBalance())
+                    .to.be.equal(voucherInitialCreditBalance)
+                    .to.be.equal(await getVoucherBalanceOnIexecPoco())
+                    .to.be.equal(voucherInitialSrlcBalance);
+                expect(await getRequesterBalanceOnIexecPoco()).to.be.equal(
+                    requesterInitialSrlcBalance - dealPrice,
+                );
+                expect(await voucher.getSponsoredAmount(dealId)).to.equal(0);
+            });
+
+            it('Should match orders boost with an authorized account', async () => {
+                for (const asset of [app, dataset, workerpool]) {
+                    await voucherHubWithAssetEligibilityManagerSigner
+                        .addEligibleAsset(voucherType, asset)
+                        .then((x) => x.wait());
+                }
+                const authorizationTx = await voucherWithOwnerSigner.authorizeAccount(
+                    anyone.address,
+                );
+                await authorizationTx.wait();
+
+                const voucherWithAnyoneSigner = voucher.connect(anyone);
```

Comment:
Move to top & reuse everywhere or direct call `voucher.connect(anyone).blabla(..)`?

Reply:
Moved to top here, will be re-used when all concerned functions will be onlyAthorized etc

Reply:
Ok I see!